extensions/ccui/uiwidgets/UIButton.js

```diff
@@ -896,7 +892,11 @@ ccui.Button = ccui.Widget.extend(/** @lends ccui.Button# */{
 
     setColor: function(color){
         cc.ProtectedNode.prototype.setColor.call(this, color);
-        this._updateTexturesRGBA();
+        this._updateTexturesColor();
+    },
+    
+    setOpacity: function(opacity){
+        cc.ProtectedNode.prototype.setOpacity.call(this, opacity);
```

Comment:
You are right, so they need to call setOpacity to modify the color, but they will find that won't have any effect, because none of the textures have modified their opacity